tests/merkle_tree_tests.move

```diff
@@ -2,9 +2,71 @@
 #[test_only]
 module bitcoin_spv::merkle_tree_tests;
 
-use bitcoin_spv::merkle_tree::{verify_merkle_proof, EInvalidMerkleHashLengh};
+use bitcoin_spv::merkle_tree::{verify_merkle_proof, merkle_hash, EInvalidMerkleHashLengh};
+use std::hash::sha2_256;
 use std::unit_test::assert_eq;
 
+fun single_hash(x: vector<u8>, y: vector<u8>): vector<u8> {
+    let mut z = x;
+    z.append(y);
+    sha2_256(z)
+}
+
+fun reduce_double_hashes(v: vector<vector<u8>>): vector<vector<u8>> {
+    let mut result = vector[];
+
+    let mut i = 0;
+    while (i < v.length() - 1) {
+        result.push_back(merkle_hash(v[i], v[i + 1]));
+        i = i + 2;
+    };
+    result
+}
+
+fun reduce_single_hashes(v: vector<vector<u8>>): vector<vector<u8>> {
+    let mut result = vector[];
+
+    let mut i = 0;
+    while (i < v.length() - 1) {
+        result.push_back(single_hash(v[i], v[i + 1]));
+        i = i + 2;
+    };
+    result
+}
+
+fun create_proof_for_testing(
```

Comment:
lets add a comment what proof we are constructing here (its our custom one) and maybe add a comment that the double_hashes are the tx_ids, its not clear

Reply:
I update the name

Reply:
I remember that if the module is #[test_only], then any function inside this is also test_only. Let's me double check this.

Reply:
> I remember that if the module is #[test_only], then any function inside this is also test_only. Let's me double check this.

Yes, in #[test_only] module, this module doesn't deploy and we don't need add #[test_only] for sub function inside this.